### ISSUE TITLE
There seems to be an error in Vaadin Grid: rowData.item is undefined

### ISSUE THREAD

**Issue by HayoBaanAtHand** [vaadin/flow-components#3730]
### Description

I updated from Vaadin 14.8.15 to Vaadin 23.2 and now I get this (see below) error (in the browser) when opening a form with grids that have drag and drop enabled.

Interestingly: it only happens the first time I open the form, and only when the first opening populated the form with an entity that had items in the drag&drop grid. If I open the form for an entity that does not have items for the d&d grid, all consecutive openings work.

Note the form is part of a layout with a main grid where you can select an entity for which the form is to be opened.

Firefox:
![Screen Shot 2022-09-16 at 10 08 43](https://user-images.githubusercontent.com/68533277/190589818-ef8c2e28-3ad7-4231-b653-818c1919d2bb.png)
Safari:
![Screen Shot 2022-09-16 at 10 17 26](https://user-images.githubusercontent.com/68533277/190591290-e34f6ca6-d7c2-434f-b74f-b1c998fcaca4.png)

### Expected outcome

No errors.

### Minimal reproducible example

I can't share the application code, so hopefully you can reproduce yourself; otherwise I will have to build a specific application to show this.

### Steps to reproduce

1. Open layout with entity selection grid as well as a form (with another item grid that has d&d enabled) to edit the entity.
2. Select an entity with items for the form grid.
3. Form opens with selected entity -> error shown in browser console.
4. Select another entity -> no error.
Note: if you first select an entity without items for the form grid, no error is produced, not even when then selecting an entity that has items for the item grid. So this looks like an initialization issue to me.

### Environment

Vaadin versions: 23.2
OS: MacOS 12.6

### Browsers

Firefox, Safari

**Comment by web-padawan** [comment]
Thanks for the issue. This seems to be the same as #2627. Could you please check if you have some code similar to the example provided in https://github.com/vaadin/flow-components/issues/2627#issuecomment-1075526179? If not, please try to provide a minimal code example to reproduce.

**Comment by HayoBaanAtHand** [comment]
The errors we are both seeing are indeed very similar. I don't hide the grid itself though. The form on which the grid is placed, however is shown (after you select an item in the entity grid). I'll try to create a small application that exhibits the issue nevertheless.

**Comment by HayoBaanAtHand** [comment]
I've been able to create an example application that can be used to reproduce the error.

1. Run application from attached zip e.g. with SPRING_PROFILES_ACTIVE=native mvn -U spring-boot:run -Dspring-boot.run.arguments='--spring.config.location=file:config/bootstrap.properties'.
2. Open browser at localhost:8080/management
3. Enter some name and password, “login”, navigate to entity page.
4. Click on the first entry in the entity grid to select it and open the editor form.
5. You'll see the error in your browser's console log.
6. If you now click any other entity in the entity select grid, no further further errors are show.
7. Reset the entity view+form by navigating back to “home” and then navigate back to the entity page.
8. Repeat steps 4 to 7, but now select the second entity (without items) -> no errors in the console log
9. Reset the entity view+form by navigating back to “home” and then navigate back to the entity page.
10. Repeat steps 4 to 7 -> again an error in the console log.

So: it is an error only after a fresh creation of the form. Hope this helps finding (and fixing :wink:) the bug.
[Grid-Bug.zip](https://github.com/vaadin/flow-components/files/9583050/Grid-Bug.zip)

**Comment by web-padawan** [comment]
Closing as duplicate of https://github.com/vaadin/flow-components/issues/2627 fixed by https://github.com/vaadin/flow-components/pull/5250.
